src/java/org/apache/cassandra/utils/ExpMovingAverage.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.utils;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.util.concurrent.AtomicDouble;
+
+/**
+ * Sample-based exponential moving average. On every update a fraction of the current average is replaced by the new
+ * sample. New values have greater representation in the average, and older samples' effect exponentially decays with
+ * new data.
+ */
+public class ExpMovingAverage implements MovingAverage
+{
+    /** The ratio of decay, between 0 and 1, where smaller alpha means values are averaged over more samples */
+    private final double alpha;
+
+    /** The long term average with exponential decay */
+    private final AtomicDouble average = new AtomicDouble(Double.NaN);
+
+    /**
+     * Create a {@link ExpMovingAverage} where older values have less than 1% effect after 1000 samples.
+     */
+    public static MovingAverage decayBy1000()
+    {
+        return new ExpMovingAverage(0.0046);
+    }
+
+    /**
+     * Create a {@link ExpMovingAverage} that averages over a minute.
```

Comment:
The JavaDoc is outdated here and below.

Reply:
Fixed in cd6fa21f342e74bb3e46d9571e26482d8e0ac215.